### PR TITLE
Loki: Add aggregate by label support

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Adds the label param handling and `by` and `without` aggregation operation variants. 
* [x] Sharing code with Prometheus builder
* [x] Need to update LabelParamEditor to support loki as well, or create a specific component for Loki

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/querybuilder/components/LabelParamEditor.tsx [hunk 1]
```diff
@@ -44,15 +45,28 @@ export function LabelParamEditor({
 
 async function loadGroupByLabels(
   query: PromVisualQuery,
-  datasource: PrometheusDatasource
+  datasource: DataSourceApi
 ): Promise<Array<SelectableValue<any>>> {
-  const labels = [{ label: '__name__', op: '=', value: query.metric }, ...query.labels];
-  const expr = promQueryModeller.renderLabels(labels);
+  if (datasource instanceof PrometheusDatasource) {
+    const labels = [{ label: '__name__', op: '=', value: query.metric }, ...query.labels];
```
Comment (aocenas) [inline public/app/plugins/datasource/prometheus/querybuilder/components/LabelParamEditor.tsx:51]: Is this line the only difference here? Could we make the `if` only on this variable?

File: public/app/plugins/datasource/prometheus/querybuilder/aggregations.ts [hunk 2]
```diff
@@ -126,14 +141,18 @@ function getAggregationWithoutRenderer(aggregation: string) {
 /**
  * Very simple poc implementation, needs to be modified to support all aggregation operators
  */
-function getAggregationExplainer(aggregationName: string, mode: 'by' | 'without') {
+function getAggregationExplainer(aggregationName: string, mode: 'by' | 'without' | '') {
```
Comment (aocenas) [inline public/app/plugins/datasource/prometheus/querybuilder/aggregations.ts:144]: nit: Maybe making the mode optional as `mode?: 'by' | 'without'` would be a bit nicer.

**Review by aocenas — APPROVED** [review]
Left small comments but otherwise looks good.

File: public/app/plugins/datasource/prometheus/querybuilder/aggregations.ts (hunk 2 above)
Comment (torkelo) [reply public/app/plugins/datasource/prometheus/querybuilder/aggregations.ts:144]: well figured the mode is by, without or empty (ie no extra clause), if mode is optional, might signify that it has some default?

Comment (torkelo) [reply public/app/plugins/datasource/prometheus/querybuilder/aggregations.ts:144]: well figured the mode is by, without or empty (ie no extra clause), if mode is optional, might signify that it has some default?

Comment (torkelo) [reply public/app/plugins/datasource/prometheus/querybuilder/aggregations.ts:144]: yea maybe mode is bad name for that param, maybe "labelClause" or something is better

**Pull request merged**